src/src/tx_main.cpp

```diff
@@ -1027,16 +1023,19 @@ static void setupTarget()
   setupTxBackpack();
 }
 
-void setup()
+bool setupHardwareFromOptions()
 {
-  bool hardware_success = true;
   #if defined(TARGET_UNIFIED_TX)
-  TxBackpack = new HardwareSerial(1);
-  ((HardwareSerial *)TxBackpack)->begin(460800, SERIAL_8N1, 3, 1);
-  SPIFFS.begin(true);
-  hardware_success = options_init();
-  if (!hardware_success)
+  if (!options_init())
   {
+    #if defined(PLATFORM_ESP32)
+    TxBackpack = new HardwareSerial(1);
+    ((HardwareSerial *)TxBackpack)->begin(460800, SERIAL_8N1, 3, 1);
+    #else
+    TxBackpack = new HardwareSerial(0);
+    ((HardwareSerial *)TxBackpack)->begin(460800, SERIAL_8N1);
+    #endif
+
```

Comment:
The reason this was outside the if block was that I could then add some debugging code to the options_init and hardware_init functions when things go wrong.

Reply:
I figured it was for having the logger available when doing the unpacking. I thought it might make sense to keep it because if it dies there, there's no UART available, but now that I am thinking about it, for it to be recovered from that state, it would need the CRSF started, not the Backpack. Should I just remove it entirely then? Or move it before the `options_init()` and comment it out maybe?

Reply:
Maybe move it into `options_init` and wrap it in a DEBUG_INIT or some such?